src/main/cpp/util/InclusiveRange.h

```diff
@@ -0,0 +1,33 @@
+#ifndef INCLUSIVE_RANGE_H_INCLUDED
+#define INCLUSIVE_RANGE_H_INCLUDED
+
+#include <boost/property_tree/ptree.hpp>
+
+namespace stride {
+namespace util {
+
+template <typename T>
+struct InclusiveRange
+{
+	InclusiveRange() : InclusiveRange(T(), T()) {}
+	InclusiveRange(T min, T max) : minimum(min), maximum(max) {}
+
+	// Parse an inclusive range from a ptree with "minimum" and "maximum" keys.
+	// Throws `boost::property_tree::ptree_error` on invalid inputs.
+	void parse(const boost::property_tree::ptree& pt)
+	{
+		minimum = pt.get<T>("minimum");
+		maximum = pt.get<T>("maximum");
+	}
+
+	T average() const { return (minimum + maximum) / 2; }
+	bool include(T x) const { return minimum <= x && x <= maximum; }
```

Comment:
Wouldn't it make more sense to rename this function `includes`? `include` sounds an awful lot like a command instead of a query.

Also, some doc comments would be nice I guess.